### ISSUE TITLE
Get full coverage in our bench marks project.

### ISSUE THREAD

**Issue by glennawatson** [reactiveui/ReactiveUI#1734]
We should get a lot more coverage of different performance tests in our Benchmark project. So most major classes covered.

## Extensions

- [ ] [When Activated](https://github.com/reactiveui/ReactiveUI/blob/4ac16d3fc703929f8a573eb1300123c5ab7f6518/src/ReactiveUI/Activation/ViewForMixins.cs#L46)
- [ ] [When Any](https://github.com/reactiveui/ReactiveUI/blob/4ac16d3fc703929f8a573eb1300123c5ab7f6518/src/ReactiveUI/VariadicTemplates.cs#L71)
- [ ] [When Any Value](https://github.com/reactiveui/ReactiveUI/blob/4ac16d3fc703929f8a573eb1300123c5ab7f6518/src/ReactiveUI/VariadicTemplates.cs#L54)
- [ ] [When Any Observable](https://github.com/reactiveui/ReactiveUI/blob/4ac16d3fc703929f8a573eb1300123c5ab7f6518/src/ReactiveUI/VariadicTemplates.cs#L1234)
- [ ] [When Any Dynamic](https://github.com/reactiveui/ReactiveUI/blob/4ac16d3fc703929f8a573eb1300123c5ab7f6518/src/ReactiveUI/VariadicTemplates.cs#L87)

## Interactions

- [ ] [Interaction](https://github.com/reactiveui/ReactiveUI/blob/4ac16d3fc703929f8a573eb1300123c5ab7f6518/src/ReactiveUI/Interactions/Interaction.cs#L50)
- [ ] [Register Handlers](https://github.com/reactiveui/ReactiveUI/blob/f49cc403c373ca89ac499d2b21f857d6876b88c0/src/ReactiveUI/Interactions/Interaction.cs#L84)
- [ ] [Handle](https://github.com/reactiveui/ReactiveUI/blob/f49cc403c373ca89ac499d2b21f857d6876b88c0/src/ReactiveUI/Interactions/Interaction.cs#L169)
- [ ] [Interaction Context](https://github.com/reactiveui/ReactiveUI/blob/f49cc403c373ca89ac499d2b21f857d6876b88c0/src/ReactiveUI/Interactions/InteractionContext.cs#L56)

## Observable Properties

- [ ] [Reactive Object](https://github.com/reactiveui/ReactiveUI/blob/master/src/ReactiveUI/ReactiveObject/ReactiveObject.cs)
- [ ] [Observable As Property Helper](https://github.com/reactiveui/ReactiveUI/blob/f49cc403c373ca89ac499d2b21f857d6876b88c0/src/ReactiveUI/ObservableForProperty/ObservableAsPropertyHelper.cs#L91)

## Reactive Commands

- [x] [Create](https://github.com/reactiveui/ReactiveUI/blob/4ac16d3fc703929f8a573eb1300123c5ab7f6518/src/ReactiveUI/ReactiveCommand/ReactiveCommand.cs#L86)
- [x] [Create From Task](https://github.com/reactiveui/ReactiveUI/blob/4ac16d3fc703929f8a573eb1300123c5ab7f6518/src/ReactiveUI/ReactiveCommand/ReactiveCommand.cs#L274)
- [x] [Create From Observable](https://github.com/reactiveui/ReactiveUI/blob/4ac16d3fc703929f8a573eb1300123c5ab7f6518/src/ReactiveUI/ReactiveCommand/ReactiveCommand.cs#L358)
- [ ] [Create Combined](https://github.com/reactiveui/ReactiveUI/blob/4ac16d3fc703929f8a573eb1300123c5ab7f6518/src/ReactiveUI/ReactiveCommand/ReactiveCommand.cs#L513)
- [ ] [Invoke](https://github.com/reactiveui/ReactiveUI/blob/4ac16d3fc703929f8a573eb1300123c5ab7f6518/src/ReactiveUI/ReactiveCommand/ReactiveCommandMixins.cs#L28)

**Comment by Kavignon** [comment]
Lately, I’ve been interested by everything performance so this could be a fun thing! are we talking of adding micro-benchmarks with a benchmark.NET? What’s the scope?

**Comment by glennawatson** [comment]
@kavignon if you want to help out this is probably a good starter task. Essentially we want a base line of most of our existing code base so we aren't going backwards in terms of performance. There is not a platform runner at the moment for benchmark dot net for Android iOS so I would focus on core functionality first that didn't exist in the platforms folder and work from there.

**Comment by Kavignon** [comment]
You mention full coverage. How do I determine case coverage to make sure when I submit my PR, it makes sense with what you’d expect in the benchmark project?

**Comment by glennawatson** [comment]
Yeah coverage is a bit tricky since as far as I know there isn't any automated tools. I was just going to go through each file in the src/ReactiveUI folder and produce a benchmark for it. I wouldn't do the Platform folder

**Comment by Kavignon** [comment]
How about this idea; I’m going to close ReactiveUI and do a few benchmarks on my own. Then submit a PR for those and we can discuss on said PR what you would have like to see if I miss anything. When that portion is stable, I’ll get over the items I haven’t done beforehand. Sounds good?

**Comment by RLittlesII** [comment]
@Kavignon  I'd recommend starting with one of the items above, there are 4 categories of benchmarks.  I believe I have the `ReactiveCommand` started.  Let me dust that branch off this weekend and submit a PR for it.  Then you can just follow that PR for the remaining benchmarks.  That way you don't invest too much effort. I'd recommend open a PR per category to keep it easier to manage.  The bigger the PR the harder it is to review.

**Comment by Kavignon** [comment]
That sounds pretty good in fact! let me know here when your PR is ready!

**Comment by RLittlesII** [comment]
Adding `ToProperty` extension methods to the list.

**Comment by GmausDev** [comment]
Hi may i can help , still open?

**Comment by glennawatson** [comment]
@GmausDev definitely one we could use help with

**Comment by glennawatson** [comment]
Closing this issue as part of a cleanup of long-standing issues. Benchmark coverage is an ongoing effort and the benchmark project has evolved since this was opened. If you'd like to contribute benchmarks for specific areas, PRs are always welcome. Thank you for the suggestion.
